### PR TITLE
Rewrite manpage

### PR DESCRIPTION
**Pull request by The-King-of-Toasters**
- Much of the existing text was reworded for better understandability.
- The new manpage uses mdoc(7) for better formatting.
  Because of this, we can merge bemenu and bemenu-run into a single
  file.

### CODE REVIEW

File: man/bemenu.1.in [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+Override the rendering scale factor for the GUI backends.
+.El
+.Sh EXIT STATUS
+.Nm
+exits 0 on success, 1 if the user exited without selecting or an error,
+and if exiting with a custom error code, 10 plus the number specified.
+.Sh AUTHORS
+.Nm
+is written an maintained by
+.An Jari Vetoniemi Aq Mt jari.vetoniemi@indooratlas.com .
```
Comment (Cloudef) [inline man/bemenu.1.in:302]: you may want to use mailroxas@gmail.com here instead rather than one of the company emails I work for.

Comment (The-King-of-Toasters) [reply man/bemenu.1.in]: Done

**Comment by Cloudef** [comment]
@Earnestly do you have any comments?

**Comment by Earnestly** [comment]
The command-line options should be described under its own `OPTIONS` section.  Further the backend specific flags should also have their own subsection (e.g. `Backend options` which matches the `SYNOPSIS` notation) within the `OPTIONS` section.

Consider giving the keyboard bindings their own subsection as well such as `KEYBOARD COMMANDS` to make it visually distinct and easier to find when used as a reference.

The rest of the sections look good and the grammar is fine and explains each option clearly.  The `SYNOPSIS` is great.

Obviously this is adopting mandoc which can be awkward for describing non-C and non-shortopts libraries and commands due to its poor design of mixing semantic markup and representation.  However for bemenu it should be fine as it doesn't use GNU style longopts.

*PS: Is there any call for a `SEE ALSO` section?*

**Comment by The-King-of-Toasters** [comment]
oh yeah, I forgot the see also sections. lemme fix that

**Comment by The-King-of-Toasters** [comment]
> The command-line options should be described under its own `OPTIONS` section. Further the backend specific flags should also have their own subsection (e.g. `Backend options` which matches the `SYNOPSIS` notation) within the `OPTIONS` section.

The BSDs like to list the options like:

> The options are as follows:
> [list]

But if you want them under their own heading, that's fine.

I noticed that color support isn't available under ncurses, so I'll add a note about that.
 
> Consider giving the keyboard bindings their own subsection as well such as `KEYBOARD COMMANDS` to make it visually distinct and easier to find when used as a reference.

Done

> Obviously this is adopting mandoc which can be awkward for describing non-C and non-shortopts libraries and commands due to its poor design of mixing semantic markup and representation. However for bemenu it should be fine as it doesn't use GNU style longopts.

You can actually get around the long opt issue by doing:

```mdoc
.Fl -longopt Ar arg
.Fl -longopt Ns Cm = Ns Ar arg
```

File: man/bemenu.1.in [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+.Ar prompt
+text to be displayed to the left of the input field.
+Defaults to
+.Ql bemenu
+.It Fl v , -version
+Print the
+.Nm
+version number to standard output and exit.
+.It Fl w , -wrap
+Wraps cursor selection.
```
Comment (The-King-of-Toasters) [inline man/bemenu.1.in:84]: I tried testing this and I still don't know what this does, so I left it as is.

File: man/bemenu.1.in [hunk 3]
```diff
@@ -0,0 +1,311 @@
...
+.Rs
+.%T Pango Reference Manual
+.%D August 20, 2020
+.%U https://developer.gnome.org/pango/1.46/
+.%O Fonts \(em pango_font_description_from_string()
+.Re
+.Sh AUTHORS
+.Nm
+is written and maintained by
+.An Jari Vetoniemi Aq Mt mailroxas@gmail.com .
```
Comment (The-King-of-Toasters) [inline man/bemenu.1.in:311]: Feel free to add any other major contributors.

Comment (Cloudef) [reply man/bemenu.1.in:311]: Might be better as separate contributors file. The current list of contributors can be seen with `git shortlog -s -n --all --no-merges`. Anyhow, don't have to include that in scope of this PR.

File: man/bemenu.1.in (hunk 2 above)
Comment (Cloudef) [reply man/bemenu.1.in:84]: The `-w` option should make the list go back to first item once you go one over the end. Thus "wrapping around".

**Comment by Earnestly** [comment]
> The BSDs like to list the options like:
The BSDs are fairly inconsistent about this.  Generally I follow the format laid out by Kerrisk in https://man7.org/linux/man-pages/man7/man-pages.7.html. It is well thought out and works well for quick reference due to better visual categorisation.

> You can actually get around the long opt issue by doing:
Yeah, I know there are many such hacks.

**Comment by Cloudef** [comment]
This looks good to me. I'll let @Earnestly approve it since he wrote the original man pages.

**Comment by Earnestly** [comment]
Overall very good, would just like to see an `OPTIONS` section and it'd be perfect. It would be better to also have the keyboard commands as their own section as well instead of being a subsection under `DESCRIPTION`.  This is all about making it easy to navigate and locate references quickly.

File: man/bemenu.1.in [hunk 4]
```diff
@@ -0,0 +1,314 @@
...
+.It Aq Cm delete
+Remove character at the cursor.
+.It Cm C-u , S- Ns Aq Cm delete
+Rrmove all characters before the cursor.
+.It Cm C-k
+Remove all characters after the cursor.
+.It Cm C-w
+Clear the filter.
+.It Cm M- Ns Op 0\(en9
+Print selected items and exit with a custom error code.
```
Comment (The-King-of-Toasters) [inline man/bemenu.1.in:269]: I also couldn't trigger this on any of the backends using either alt key.

**Comment by The-King-of-Toasters** [comment]
I think we're almost done. You can see how it looks in [HTML](https://man.sgregoratto.me/bemenu.1) and [PDF](https://www.sgregoratto.me/paste/bemenu.pdf) form.

File: man/bemenu.1.in (hunk 4 above)
Comment (Cloudef) [reply man/bemenu.1.in:269]: I think that only works with wayland and x11. Also needs bemenu from git.

File: man/bemenu.1.in [hunk 5]
```diff
@@ -0,0 +1,314 @@
...
+.It Fl i , -ignorecase
+Filter items case-insensitively.
+.It Fl l Ar number , Fl -list Ar number
+List items vertically with the given
+.Ar number
+of lines.
+.It Fl P Ar prefix , Fl -prefix Ar prefix
+Display
+.Ar prefix
+before the highlighed item in a vertical list.
```
Comment (Earnestly) [inline man/bemenu.1.in:72]: Typo here on `highlighed` which should be `highlighted` (throughout the manual)

**Review by Earnestly — COMMENTED** [review]
I don't get this github UI, anyway there's still two cases of `Highlighed`:

```
Highlighed background.
Highlighed foreground.
```

**Comment by Earnestly** [comment]
Everything seems pretty good now imo; it is a nice improvement over the last effort.

Though I do wonder if @Cloudef wants `--disco` documented :P

**Comment by Cloudef** [comment]
Maybe remove that since it's considered to be "easter egg" :)

**Comment by The-King-of-Toasters** [comment]
done

**Comment by Cloudef** [comment]
Merged, thanks!

**Pull request merged**